### PR TITLE
6096: Add SMTP Healthchecks to ISAPI

### PR DESCRIPTION
**Pull request by assoomro**
- added hchecks
- refactored db healthchecks - now we control the database state, instead of relying on a 'broken' copy
- moved 'RegisterHealthChecks' extension to 'Common' project

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Identity.Common/NHSD.BuyingCatalogue.Identity.Common.csproj [hunk 1]
```diff
@@ -18,9 +18,19 @@
...
       <_Parameter1>NHSD.BuyingCatalogue.Identity.Api.UnitTests</_Parameter1>
     </AssemblyAttribute>
+
+    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
+      <_Parameter1>NHSD.BuyingCatalogue.Identity.Api</_Parameter1>
+    </AssemblyAttribute>
+
+    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
+      <_Parameter1>NHSD.BuyingCatalogue.Organisations.Api</_Parameter1>
+    </AssemblyAttribute>
```
[the remark this answers is not in the document]
Comment (assoomro) [reply src/NHSD.BuyingCatalogue.Identity.Common/NHSD.BuyingCatalogue.Identity.Common.csproj:28]: Guess I overthought this one a bit, I'll revert it to having a public access modifier :]

File: src/NHSD.BuyingCatalogue.Identity.Common/Extensions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 using NHSD.BuyingCatalogue.Identity.Common.Constants;
 using NHSD.BuyingCatalogue.Identity.Common.Settings;
 
-namespace NHSD.BuyingCatalogue.Organisations.Api.Extensions
+namespace NHSD.BuyingCatalogue.Identity.Common.Extensions
 {
     internal static class ServiceCollectionExtensions
```
[the remark this answers is not in the document]
Comment (assoomro) [reply src/NHSD.BuyingCatalogue.Identity.Common/Extensions/ServiceCollectionExtensions.cs:10]: As above ^

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/DatabaseSteps.cs [hunk 3]
```diff
@@ -8,10 +8,12 @@ namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
     public sealed class DatabaseSteps
     {
         private readonly Settings _settings;
+        private readonly ScenarioContext _context;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/DatabaseSteps.cs:11]: Do we need this? Other than being assigned in the constructor it doesn't look like we're using it anywhere.

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/DatabaseSteps.cs [hunk 4]
```diff
@@ -8,10 +8,12 @@ namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
...
 
-        public DatabaseSteps(Settings settings)
+        public DatabaseSteps(Settings settings, ScenarioContext context)
         {
             _settings = settings;
+            _context = context;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/DatabaseSteps.cs:16]: As above – do we need this?

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/DatabaseSteps.cs (hunk 3 above)
Comment (assoomro) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/DatabaseSteps.cs]: Great spot! I totally forgot to remove that. Thanks!

**Pull request merged**